### DISCUSSION TITLE
Set and remove MachineEnvironments with a little Function

### DISCUSSION THREAD

**Discussion by StHoBT** [PSAppDeployToolkit/PSAppDeployToolkit#772] [Q&A] [answered]
I have developed two functions to set and delete machine environment variables.

Here are the two functions:

Function Set-MachineEnvironment{
<#
.SYNOPSIS
	Add an Entry to the Machine-Environment
.DESCRIPTION
	with this function is it possible to add an Entry to the Machine-Environment.
.SYNTAX
    Set-MachineEnvironment [-Name] <String> [-Value] <String> [[-Debuglevel] <Int>] 
.PARAMETER Name
	the String of the Entry that is to add. -Name "TestSHO"
.PARAMETER Value
	the String of the Value of the Entry that is to add. -Value "Lic-Server@test.de"
.PARAMETER Debuglevel
	if the debuglevel is higher then 0 debuginformations will be written in the 'Bertrandt Custom Log'. The Parameter is optional. default=0
	0=no debug , 1=only ReturnValue (no ReturnValue is present) , 2=full debug with most values
.EXAMPLE
	Set-MachineEnvironment "TestSHO" "Lic-Server@test.de" -Debuglevel 2
.NOTES
	Attention: You must use the Full right String. Asterisk are not allowed
	Author of this feature: Steffen Hoffmann
#>
	Param(
	[Parameter(Mandatory=$true)][String]$Name,
	[Parameter(Mandatory=$true)][String]$Value,
	[Int]$Debuglevel=0
	)
	If($Debuglevel -ge 1){
		Write-Log -Message "Add a Entry to the Machine-Environment" -Source 'Bertrandt Custom Log' -LogType 'CMTrace'
		Write-Log -Message "Name=$Name" -Source 'Bertrandt Custom Log' -LogType 'CMTrace'
		Write-Log -Message "Value:$Value" -Source 'Bertrandt Custom Log' -LogType 'CMTrace'
	}
	
	[System.Environment]::SetEnvironmentVariable($Name, $Value, 'Machine')
	If($Debuglevel -ge 1){
		Write-Log -Message "The New Machine-Environment $Name is set to $Value" -Source 'Bertrandt Custom Log' -LogType 'CMTrace'
	}
}

Function Remove-MachineEnvironment{
<#
.SYNOPSIS
	Remove an Entry from the Machine-Environment
.DESCRIPTION
	with this function is it possible to Remove an Entry from the Machine-Environment.
.SYNTAX
    Remove-MachineEnvironment [-Name] <String> [[-Debuglevel] <Int>] 
.PARAMETER Name
	the String of the Entry that is to remove. -Name "TestSHO"
.PARAMETER Debuglevel
	if the debuglevel is higher then 0 debuginformations will be written in the 'Bertrandt Custom Log'. The Parameter is optional. default=0
	0=no debug , 1=only ReturnValue (no ReturnValue is present) , 2=full debug with most values
.EXAMPLE
	Remove-MachineEnvironment "TestSHO" -Debuglevel 2
.NOTES
	Attention: You must use the Full right String. Asterisk are not allowed
	Author of this feature: Steffen Hoffmann
#>
	Param(
	[Parameter(Mandatory=$true)][String]$Name,
	[Int]$Debuglevel=0
	)
	If($Debuglevel -ge 1){
		Write-Log -Message "Remove a Entry from the Machine-Environment" -Source 'Bertrandt Custom Log' -LogType 'CMTrace'
		Write-Log -Message "Name=$Name" -Source 'Bertrandt Custom Log' -LogType 'CMTrace'
	}
	
	[System.Environment]::SetEnvironmentVariable($Name, $Null, 'Machine')
	If($Debuglevel -ge 1){
		Write-Log -Message "The Machine-Environment $Name is removed" -Source 'Bertrandt Custom Log' -LogType 'CMTrace'
	}
}

**Comment by mjr4077au** [comment, marked as answer]
Functionality to achieve this has been added via https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/commit/d7b68a495d74a4b25343b54a9bac1ad12febdbc1, and will be part of PSAppDeployToolkit 4.1.0. Thank you for your time and contributions on multiple discussions!
